src/period.cpp

```diff
@@ -8,6 +8,14 @@
 #include "nanotime/pseudovector.hpp"
 #include "nanotime/utilities.hpp"
 
+// See comment in <R_Ext/Complex.h> in R>=4.3.0. As of 2024-10-30
+//   clang trunk with -Wmissing-braces requires the double-braced
+//   approach for creating Rcomplex objects.
+#if R_VERSION >= R_Version(4, 3, 0)
+#define RCOMPLEX(RE, IM) Rcomplex{{RE, IM}}
+#else
+#define RCOMPLEX(RE, IM) Rcomplex{RE, IM}
+#endif
```

Comment:
That's not code I particularly like.

Reply:
I have almost the same code here now.  You are missing a leading `inline` (for slighly better performance / no function call overhead).

Reply:
I see in your comment https://github.com/eddelbuettel/nanotime/issues/134#issuecomment-2448459585 you branched on `#ifdef R_LEGACY_RCOMPLEX`. Nice idea!

I think relying on that assumes R>=4.3.0, though, so I don't want to make it the primary branch. OTOH, _only_ branching on R version prevents any user that might need to use the legacy definition (for some reason?) from doing so. So I'll try and use both.